### PR TITLE
[6239] fix for mangled subcollection name upon rename (main)

### PR DESCRIPTION
**Pull request by tsmeele**
Use RDBMS internal function to calculate stringlength as input to substr() operations, rather than relying on a client-calculated literal. This avoids differences in interpretation of string length related to unicode characters that span multiple bytes.

### CODE REVIEW

**Comment by trel** [comment]
is this safe across mysql/mariadb and postgresql?

**Comment by tsmeele** [comment]
The length() function is standard across SQL implementations.  Note that chosen string encoding in the RDBMS influences the actual result returned by length().  The proposed solution takes advantage of the alignment of length(), substr() and string encoding within the RDBMS of the vendor, rather than requiring the client to anticipate on differences across vendors and encodings.

**Comment by SwooshyCueb** [comment]
> The proposed solution takes advantage of the alignment of length(), substr() and string encoding within the RDBMS of the vendor, rather than requiring the client to anticipate on differences across vendors and encodings.

I certainly like the sound of that

**Comment by trel** [comment]
Excellent.

I see clang-format has an opinion...

Otherwise, I think this should turn green on the next push.

**Review by korydraughn — COMMENTED** [review]
Very nice.

We'll need to add a test based on this: https://github.com/irods/irods/issues/6239#issuecomment-2535223519

We'll take care of that.

File: plugins/database/src/db_plugin.cpp [hunk 1]
```diff
@@ -11375,17 +11375,17 @@ irods::error db_move_object_op(
         len = strlen( collNameSlash );
         snprintf( collNameSlashLen, 10, "%d", len );
         cllBindVars[cllBindVarCount++] = newCollName;
-        cllBindVars[cllBindVarCount++] = ocLenStr;
+        cllBindVars[cllBindVarCount++] = oldCollName;
         cllBindVars[cllBindVarCount++] = newCollName;
-        cllBindVars[cllBindVarCount++] = ocLenStr;
-        cllBindVars[cllBindVarCount++] = collNameSlashLen;
```
Comment (alanking) [inline plugins/database/src/db_plugin.cpp:11381]: We might be able to remove `ocLenStr` and `collNameSlashLen` as they appear to only be used for this query.

Comment (tsmeele) [reply plugins/database/src/db_plugin.cpp:11381]: Good point, I have added a change to remove the unused declarations related to length calculation

**Review by alanking — COMMENTED** [review]
Looks good. I'll try to get a test written for this so we can get this in ASAP

**Comment by alanking** [comment]
Just wanted to echo the comment I made in https://github.com/irods/irods/issues/6239#issuecomment-2551869538...

After trying this out manually and writing some automated tests, it appears that the issue concerning renaming collections with multi-byte characters in the path remains even with these changes. Will leave this open while we look for another solution...

**Comment by tsmeele** [comment]
The method in principle should fix the issue, yet I had overlooked a subtle increment of the old ocLen during conversion to string.  Now added the increment to the SQL.   You may want to add a test without multi-byte as well to ensure behavior is correct on single-byte unicode characters as well.

**Comment by korydraughn** [comment]
Sounds good. We'll confirm things work as intended.
